x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/NativePrivilegeStore.java

```diff
@@ -133,25 +195,75 @@ public void getPrivileges(Collection<String> applications, Collection<String> na
                         .setFetchSource(true)
                         .request();
                     logger.trace(() ->
-                        new ParameterizedMessage("Searching for privileges [{}] with query [{}]", names, Strings.toString(query)));
+                        new ParameterizedMessage("Searching for [{}] privileges with query [{}]",
+                            applications, Strings.toString(query)));
                     request.indicesOptions().ignoreUnavailable();
+                    // TODO: not parsing source of cached entries?
```

Comment:
I don't think we need this TODO do we?
I think in this context it's simpler to parse everything than try and track doc changes (because we trust the cache if it's populated, we only go to the index when the cache is missing something).

Reply:
You are right. I wrote the Todo when I was trying out a few optimisation ideas, e.g.: 1) only fetch things that are missing in cache; 2) fetch everything but only parsing what is missing in cache;

But later we decided to simplify things by always fetching everything if we have to hit the index and repopulate the cache to make sure it is always up to date. So this todo is no longer necessary.